### PR TITLE
[5454] Allow upload of csv with blank rows

### PR DESCRIPTION
**Pull request by richardpattinson**
### Context

Submitting a CSV with a blank row leads to the error message ”  must match a trainee record”.

However the csv file should instead be accepted.

### Changes proposed in this pull request

Add a guard method to ensure that a blank row csv is accepted.

### Guidance to review

Try to submit a blank row csv.

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/services/bulk_update/recommendations_uploads/validate_csv_row.rb [hunk 1]
```diff
@@ -41,6 +41,17 @@ def validate!
...
         subject
+        if csv_row_empty?(row)
+          messages = []
+        end
+      end
+
+      def csv_row_empty?(row)
+        row.instance_variables.each do |variable|
+          value = row.instance_variable_get(variable)
+          return false unless value.nil?
```
Comment (markhallen) [inline app/services/bulk_update/recommendations_uploads/validate_csv_row.rb:52]: Using `blank?` here also covers empty strings ("").
```suggestion
          return false unless value.blank?
```

File: spec/services/bulk_update/recommendations_uploads/validate_csv_row_spec.rb
Comment (markhallen) [inline spec/services/bulk_update/recommendations_uploads/validate_csv_row_spec.rb:1]: Can we also add a test for empty strings?

**Review by markhallen — CHANGES_REQUESTED** [review]

**Comment by claire-hughez** [comment]
[TEST2.csv](https://github.com/DFE-Digital/register-trainee-teachers/files/11450719/TEST2.csv)
When I use this CSV in the deployment, it says:

3 trainees who will be recommended for QTS or EYTS
2 trainees who will not be recommended, as you did not give a date when they met the QTS or EYTS standards

However, it should not count '2 trainees' - there's no detail for them. It shouldn't be mentioned at all?
